Wikipedia/Code/WMFArticleViewController.m

```diff
@@ -1690,7 +1690,23 @@ - (void)sectionEditorFinishedEditing:(SectionEditorViewController *)sectionEdito
 
 #pragma mark - Article link and image peeking via WKUIDelegate
 
+- (BOOL)wasReadMoreItemTouchedRecently {
+    NSDate *lastReadMoreItemTouchStartDate = self.webViewController.lastReadMoreItemTouchStartDate;
+    if (lastReadMoreItemTouchStartDate != nil) {
+        NSTimeInterval timeSinceLastReadMoreItemTouchStart = [[NSDate date] timeIntervalSinceDate: self.webViewController.lastReadMoreItemTouchStartDate];
+        if (timeSinceLastReadMoreItemTouchStart < 1) {
+            return YES;
+        }
+    }
+    return NO;
+}
+
 - (BOOL)webView:(WKWebView *)webView shouldPreviewElement:(WKPreviewElementInfo *)elementInfo {
+    BOOL isReadMoreItemBeingPeeked = [self wasReadMoreItemTouchedRecently];
+    
+// TODO: add event logging here (can now know if peeked item is a `Read more` item by checking `isReadMoreItemBeingPeeked`.
+NSLog(@"\nisReadMoreItemBeingPeeked = %d\n", isReadMoreItemBeingPeeked);
```

Comment:
@nambatee Here's the flag you can now check to see if a peeked item is a `Read more`.